sycl/test-e2e/Basic/vector/operators.cpp

```diff
@@ -44,9 +44,10 @@ int main() {
 
   {
     using vec_type = s::vec<char, 1>;
-    vec_type res;
+    using res_type = s::vec<signed char, 1>;
```

Comment:
I assume this is needed because `char` and `signed char` are distinct types. However, based on the SYCL specification, I don't even see the comparison operators saying what happens with `vec<char, N>`. If we make the assumption that it is the same as for `int8_t`, then the result type will be `int8_t`, but (to my knowledge) there is no guarantee that `int8_t` is `signed char` instead of `char`.

I think I would prefer that we use `int8_t` for both the input and return type here.